### DISCUSSION TITLE
How do I include a folder containing my .cls file?

### DISCUSSION THREAD

**Discussion by GoofyPy** [James-Yu/LaTeX-Workshop#4318] [Q&A] [answered]
So I have this folder structure:

```
root
|- main.tex
|- myfolder
   |- abc.cls
```

When I want to compile main.tex:

```
\documentclass[a4paper,twoside,10pt]{abc}
\begin{document}
Hello world
\end{document}
```
it says that it can't find abc.cls. How do I fix this?

Thanks!

**Comment by jlelong** [comment]
This is not an issue with the extension but rather a LaTeX question. See the use of the variable `TEXINPUTS`, for instance https://tex.stackexchange.com/questions/93712/definition-of-the-texinputs-variable. You need to add `myfolder` to `TEXINPUTS`.

**Reply by GoofyPy** [reply]
But isn't there a extension-specific solution for this? For example, I played around with setting the environment variables (https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-tools). 

Also, I don't understand the answers from the link you provided. For example, how am I supposed to run "export TEXINPUTS=/path/to/the/local/folder//:"? Do I put that in the command line? That throws an error

**Reply by jlelong** [reply]
Folliwing https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-tools, you can add

```json
"env": {
   "TEXINPUTS": "%DIR%/myfolder"
}
```

**Reply by GoofyPy** [reply]
That works but it seems that the default value is overwritten and now it can't find article.cls. https://tex.stackexchange.com/questions/93712/definition-of-the-texinputs-variable says that the colon used in "export TEXINPUTS=/path/to/the/local/folder//:" should append the value rather replace it. but I don't see how I can achieve that within the env entry?

```
"env": {
   "TEXINPUTS": "%DIR%/myfolder//:"
}
```
does not work.

**Reply by jlelong** [reply]
It should be working. Maybe, if you are under Windows, you can try  the value `%DIR%/myfolder//;$TEXINPUTS".

**Reply by GoofyPy** [reply]
I already tried that that seems to lead to an infinite loop:
warning: kpathsea: variable `TEXINPUTS' references itself (eventually).

**Reply by GoofyPy** [reply, marked as answer]
I got it, this works:

```
"env": {
                "TEXINPUTS": ".;%DIR%//;"
            }
```

see: https://tex.stackexchange.com/questions/410350/texinputs-on-windows
